### PR TITLE
Donate updates

### LINKED ISSUE #4133: Cycle 2024.7: Donation improvements
## Must do
- [ ] Update thank you email copy
- [ ] Update donation amount defaults
- [ ] Update the minimum donation amount

## Could do
- [ ] Investigate quicker turnaround on updating donor name list
    - Particular pain point is just knowing when it’s been updated

See [the spec](https://docs.google.com/document/d/1c06WhqmTVt0HZCIR_rR1gZk87NCm09xFkpGK1B-pP9I/edit) for details.

### PR DESCRIPTION
**Pull request by rakyi**
As requested by Charlie and Hassan.

I tested the changes locally, following the instructions in the README.

Fixes #4133

### CODE REVIEW

File: functions/donation/thank-you.ts [hunk 1]
```diff
@@ -32,27 +32,79 @@ const hasThankYouEnvVars = (env: unknown): env is ThankYouEnvVars => {
 }
 
 const constructMessage = (data: MessageData): string => {
+    // This is more readable and makes the plain text formatting easier than a
+    // template string.
     return [
-        `Dear ${data.name ? data.name : "Sir/Madam"},`,
-        "Thank you for your donation to support Global Change Data Lab – the non-profit organization that helps produce the website Our World in Data.",
+        "Thank you for supporting Our World in Data!",
+        "Your generous donation to Global Change Data Lab, the nonprofit behind Our World in Data, helps us achieve our mission to make data and research on the world’s biggest challenges easier  to understand and use, for everyone: https://ourworldindata.org/problems-and-progress Together, we can make progress against those challenges.",
```
Comment (ikesau) [inline functions/donation/thank-you.ts:39]:
```suggestion
        "Your generous donation to Global Change Data Lab, the nonprofit behind Our World in Data, helps us achieve our mission to make data and research on the world’s biggest challenges easier to understand and use, for everyone: https://ourworldindata.org/problems-and-progress Together, we can make progress against those challenges.",
```

Comment (ikesau) [reply functions/donation/thank-you.ts:39]: Should "Together, we can make progress against those challenges." be on a separate line? I feel like having more text after a URL is slightly harder to read.

File: functions/donation/thank-you.ts [hunk 2]
```diff
@@ -32,27 +32,79 @@ const hasThankYouEnvVars = (env: unknown): env is ThankYouEnvVars => {
...
+        data.showOnList &&
+            "In recognition of your support, we will be delighted to include your name as part of our List of Supporters: https://ourworldindata.org/funding. We will add your name the next time we update the list, which we do every few months. The amount of your donation will not be disclosed.",
+        "Stay connected with our work: Follow us on social media or sign up for one of our newsletters here: https://ourworldindata.org/#subscribe (As a valued donor, we may also share an occasional donor-only update with you, from which you may unsubscribe at any time.)",
+        `Are you a UK taxpayer? If so, you can increase the value of your donation to us by 25% — at no extra cost to you — through the UK government’s Gift Aid scheme. Simply enter your name and address and tick a box on this form to confirm your UK taxpayer status, and we’ll take care of the rest:\n\n https://docs.google.com/forms/d/e/1FAIpQLSe7Mgm70-3UiRfh_aEJCusWCabdBFPN7hXoXMnby_6aAJsJVg/viewform?usp=pp_url&entry.2044643286=gcdl_${data.customerId}.\n\nLet us know if you have any questions: donate@ourworldindata.org`,
```
Comment (ikesau) [inline functions/donation/thank-you.ts:46]: May as well get rid of the inline `\n\n`'s here and store them as separate strings? 🚲 🖌️

Comment (rakyi) [reply functions/donation/thank-you.ts:46]: Good catch, that's a remnant from the time the whole part was conditional.

File: functions/donation/thank-you.ts (hunk 1 above)
Comment (rakyi) [reply functions/donation/thank-you.ts]: Both changed.

File: functions/donation/thank-you.ts (hunk 2 above)
Comment (rakyi) [reply functions/donation/thank-you.ts]: Changed.

**Pull request merged**